### PR TITLE
LLVM 15

### PR DESCRIPTION
**Pull request by thedmd**
Based on <https://github.com/compiler-explorer/compiler-explorer/pull/3461/commits>

Sibling of <https://github.com/compiler-explorer/infra/pull/799>

### CODE REVIEW

**Comment by dkm** [comment]
Thanks for this change. You'll also need to do something similar as https://github.com/compiler-explorer/infra/pull/691 in our infra repository. Can you take care of it ? Thanks!

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -320,6 +320,10 @@ compiler.clang1400.exe=/opt/compiler-explorer/clang-14.0.0/bin/clang++
 compiler.clang1400.semver=14.0.0
 compiler.clang1400.options=--gcc-toolchain=/opt/compiler-explorer/gcc-11.2.0
 compiler.clang1400.ldPath=${exePath}/../lib|${exePath}/../lib/x86_64-unknown-linux-gnu
+compiler.clang1500.exe=/opt/compiler-explorer/clang-15.0.0/bin/clang++
+compiler.clang1500.semver=15.0.0
+compiler.clang1500.options=--gcc-toolchain=/opt/compiler-explorer/gcc-11.2.0
```
Comment (dkm) [inline etc/config/c++.amazon.properties:325]: You can probably point to gcc-12.2.0 here

File: etc/config/c.amazon.properties [hunk 2]
```diff
@@ -251,6 +251,9 @@ compiler.cclang1301.options=--gcc-toolchain=/opt/compiler-explorer/gcc-11.2.0
 compiler.cclang1400.exe=/opt/compiler-explorer/clang-14.0.0/bin/clang
 compiler.cclang1400.semver=14.0.0
 compiler.cclang1400.options=--gcc-toolchain=/opt/compiler-explorer/gcc-11.2.0
+compiler.cclang1500.exe=/opt/compiler-explorer/clang-15.0.0/bin/clang
+compiler.cclang1500.semver=15.0.0
+compiler.cclang1500.options=--gcc-toolchain=/opt/compiler-explorer/gcc-11.2.0
```
Comment (dkm) [inline etc/config/c.amazon.properties:256]: same here

**Review by dkm — CHANGES_REQUESTED** [review]
LGTM, only minor nits, thanks

**Comment by Alcaro** [comment]
> You'll also need to do something similar as [compiler-explorer/infra#691](https://github.com/compiler-explorer/infra/pull/691) in our infra repository.

Already exists. https://github.com/compiler-explorer/infra/pull/799

**Comment by dkm** [comment]
I was expecting a link between the 2. Thanks for pointing that out!

File: etc/config/c++.amazon.properties (hunk 1 above)
Comment (thedmd) [reply etc/config/c++.amazon.properties]: Done

File: etc/config/c.amazon.properties (hunk 2 above)
Comment (thedmd) [reply etc/config/c.amazon.properties]: and done

**Comment by thedmd** [comment]
> I was expecting a link between the 2. Thanks for pointing that out!

Sorry, will keep PR's linked next time.

Bumped GCC to 12.2.0 too.

**Review by dkm — APPROVED** [review]

**Pull request merged**